index.bs

```diff
@@ -3843,6 +3844,9 @@ Verifiers may use the following algorithm to verify an encoded {{CollectedClient
 1. Append 0x2c2263726f73734f726967696e223a (`,"crossOrigin":`) to |expected|.
 1. If |crossOrigin| is true:
     1. Append 0x74727565 (`true`) to |expected|.
+    1. If |topOrigin| is defined:
+        1. Append 0x2c22746f704f726967696e223a (`,"topOrigin":`) to |expected|.
+        1. Append [=CCDToString=](|topOrigin|) to |expected|.
```

Comment:
This is complicated a bit by there existing both L2 and L3 client implementations at the same time, where L2 clients may include `"crossOrigin":true` but not include `"topOrigin"`. This logic as currently written handles the cases asymmetrically:

- It is probably correct that an "L2" RP, expecting `crossOrigin: true, topOrigin: undefined`, should **accept** an L3 client data with `"crossOrigin":true,"topOrigin":"<something>"`, since `"topOrigin"` is an unknown attribute to that RP and so it falls under the "allow any trailing attributes" rule in the final step.

- The inverse is not necessarily correct, though: an "L3" RP expecting `crossOrigin: true, topOrigin: "<something>"` might still want to accept L2 client data with `"crossOrigin":true` (and `"topOrigin"` not present). That currently **fails** this rule.

So I think it's at least worth to somehow call out that the RP may want to retry the verification with expecting `topOrigin: undefined`, perhaps as an "If the RP wants to support L2 client implementations, retry the verification procedure with the |topOrigin| input undefined" follow-on to step 13. I don't really like the way that's going if we continue adding more attributes, but I think it's necessary.

Reply:
Hm, I just noticed that another wrinkle to this is that `crossOrigin` is not backwards-compatible in the same way in the limited verification procedure.

As noted in #1631, `crossOrigin` is defined as `optional` in the Web IDL, probably for backwards compatibility with L1 clients. But the limited verification procedure does not maintain this backwards compatibility with `"crossOrigin"` being absent in the client data.

So maybe we shouldn't attempt to keep `topOrigin` backwards-compatible either?

Reply:
Oh, right. That is an excellent point.

Reply:
I added a _requireTopOrigin_ parameter and extended the verification steps to use it to decide backwards compatibility, that resolves this review comment. I also deleted the _crossOrigin_ parameter as its value can now equivalent to _topOrigin_ being defined.